src/index.ts

```diff
@@ -32,6 +32,13 @@ export interface ECRDeploymentProps {
    */
   readonly dest: IImageName;
 
+  /**
+   * The image architecture to be copied.
+   *
+   * @default - ""
+   */
+  readonly imageArch?: string;
```

Comment:
I'd probably lean towards this. What do you think?
```suggestion
  readonly architecture?: string;
```

Reply:
`imageArch` is intentionally used to avoid any potential confusion with the architecture field referenced by `lambda.SingletonFunction` in the `ECRDeployment` class.

Currently, our Dockerfile uses a hard-coded architecture `GOARCH=amd64`.  If we'd like to allow customers to control what's used for the lambda function, we can uplift such a constraint and then expose `lambdaArch` as a sibling field of `imageArch`. 

Hope that makes sense.

Reply:
Great comment!  I've updated the PR description (copied below) to provide more clarity. 

Regarding why we don't use `string[]`, the reason is that the underlying `containers/image` lib doesn't give us such an option, as shown [here](https://github.com/containers/image/blob/main/types/types.go#L602).

Note that the current system behavior is that the destination image is always `amd64` even if it's copied from a multi-arch source image.  For example, if the multi-arch source image supports, this PR will give users the flexibility to choose between `amd64` and `arm64`, instead of having to live with `amd64`.

===========================================================================
### What's the problem?

It's observed that when copying a multi-arch source image from location A to B in ECR, we always end up with `amd64` in the destination image at location B.  The hypothesis is that the underlying lambda function used to auto-detect the architecture is backed up by Dockerfile with [a hard-coded runtime.GOARCH](https://github.com/cdklabs/cdk-ecr-deployment/blob/dcd74cdfc394d6a43a5517cb5cce1843e1fe3111/lambda/Dockerfile#L11).

### Proposed Solution
Instead of relying on an auto-detected architecture, this PR is meant to add a feature to allow users to explicitly specify a CPU architecture for the to-be-copied image, by triggering the underlying logic [here](https://github.com/containers/image/blob/5263462aa97ae23d3a620b4dcdbbc557665a42c8/internal/pkg/platform/platform_matcher.go#L164).  This will unblock the usage of Graviton instances for [Amazon SageMaker](https://aws.amazon.com/sagemaker/) and other computing platforms.

Reply:
From the point of view of a user, this documentation string is at worst lying, and at best unhelpful:

```
 @default - the underlying lambda auto-detects the relevant architecture (e.g., amd64, arm64)
   * https://github.com/containers/image/blob/main/internal/pkg/platform/platform_matcher.go#L161
```

This description is true for the *library* that we use. 

But from the point of view of a CDK user, they don't control the platform of the Lambda that runs the library. That library will always run on `amd64`. So the library will helpfully autodetect `amd64` in all cases, and there's  nothing that CDK user can do about it.

We should just state that the default if unspecified is `amd64`.

Question: what happens if the target image only has an `arm64` variant and not an `amd64` viarant, and we leave the default off? Will the library then copy the one variant that's there, or will it fail because `amd64` doesn't exist?

Reply:
> Regarding why we don't use string[], the reason is that the underlying containers/image lib doesn't give us such an option, as shown [here](https://github.com/containers/image/blob/main/types/types.go#L602).

I am looking at the documentation of this package, and I can at least see a type `ImageListSelection` that seems to indicate it's possible to copy a list of images, and a GitHub issue saying it's currently possible to [copy all architectures](https://github.com/containers/image/issues/1687).

https://pkg.go.dev/github.com/containers/image/v5/copy#Image

Do these things not actually work?